applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_vmsmonolithic.py

```diff
@@ -144,133 +133,160 @@ def PrepareModelPart(self):
         if hasattr(self, "_turbulence_model_solver"):
             self._turbulence_model_solver.PrepareModelPart()
 
-    def AddDofs(self):
-        ## Base class DOFs addition
-        super(TrilinosNavierStokesSolverMonolithic, self).AddDofs()
-
-        KratosMultiphysics.Logger.Print("DOFs for the VMS Trilinos fluid solver added correctly in all processors.")
-
-
     def Initialize(self):
-        ## Construct the communicator
-        self.EpetraCommunicator = KratosTrilinos.CreateCommunicator()
+        # If there is turbulence modelling, set the Epetra communicator in the turbulence solver
+        # TODO: Check if this needs to be done before creating the strategy
+        if hasattr(self, "_turbulence_model_solver"):
+            self._turbulence_model_solver.SetCommunicator(self.get_epetra_communicator())
+
+        # Construct and set the solution strategy
+        solution_strategy = self.GetSolutionStrategy()
+        solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
+
+        # Initialize the solution strategy
+        if not self.is_restarted():
```

Comment:
I would not just copy paste this, I am not sure if it is correct
In fact I was thinking of removing this in the structural solver

Reply:
OK. I took it from the structural solver.

Indeed, I think that the restart is nor working right now in the FluidDynamicsApplication.

Reply:
I am 99% sure it works, @AndreasWinterstein uses it constantly

hence I would not just change it

actually there is an issue related to this: #3348, but I could still not make up my mind abt what is the correct behavior :D :sweat_smile:

Reply:
I thought no one was using the restart. Is it working in all the solvers?

Reply:
I removed this if statement. Now the initialize has the same behaviour as before.
I tried the FS and Monolithic (both in serial and MPI) and the restart is saving and loading well.